### DISCUSSION TITLE
Combobox default & filter use case

### DISCUSSION THREAD

**Discussion by seanmateer** [twilio-labs/paste#1067] [Q&A] [answered]
Hi there, 

I have a use case for a combobox or select autocomplete interaction that differs slightly from the [examples](https://paste.twilio.design/components/combobox), and am wondering if it's achievable or whether there is a better pattern/component to use as an alternative. 

**Use Case:**
A SG parent (admin) account viewing their email deliverability dashboard sees stats for their parent user by default. The parent needs a way to view/filter an entire list of their subaccounts (they could have 10s or 100s) and select 1 in order to refresh stats data for that account. 

**A few requirements/constraints:**
- A user needs to see which account is selected at all times. 
- Not a multi-select use case. Only 1 account at a time. 
- A user needs to be able to filter from a preset dropdown list (like clicking the field in the [autocomplete example](https://paste.twilio.design/components/combobox#autocomplete-combobox)) 
- There will always be a default selection (assuming this is possible, but don't see an example under combobox). 
- The input should never have null value (perhaps reset to last valid selection if a user were to click out while text is empty/invalid?)

Ultimately I tried [mocking something](https://twilio.invisionapp.com/share/5XZREP52NS3#/441620253_Subuser-Account_Dropdown) inspired by the u[seCombobox state hook example](https://paste.twilio.design/components/combobox#usecombobox-state-hook), but the x-ing out feels a little unnatural and perhaps more complicated than it needs to be. 

An ideal/simpler solution might work more like this [first single selection example](https://designsystem.morningstar.com/components/combo-box/?version=4.0.2&tab=variations) from Morningstar's design system, except with the ability to set something as the default. What I like about this implementation:
- Can always see full list of results - keeps your prev selection highlighted
- You are "ready to type/filter" immediately - don't need to X anything. 
- Saves your last selection if you click away. 

Happy to bring this to DSYS office hours as a next step but wanted to see if anyone had immediate ideas for alt solutions or whether this is feasible with the current component.

**Comment by TheSisb** [comment]
Hi @seanmateer,

You can pass an initial value to the [simple combobox](https://paste.twilio.design/components/combobox#basic-combobox) with the `initialSelectedItem="Octavia Butler"` prop.

Curious if you considered using a [Select](https://paste.twilio.design/components/select) component as well?

**Comment by richbachman** [comment, marked as answer]
Hi @seanmateer 

In reference to your proposed solution, all of this should be possible but at this point we've not tested anything. 

From a UX perspective, we agree with your hunch about "x-ing" out. That can probably be dropped. Clicking into the field in order to "clear" the selection, while maintaining the previous selection feels like a good approach. 

Engineering will most likely need to use the [state hook within Combobox](https://paste.twilio.design/components/combobox#usecombobox-state-hook), since that will give them the most control of the combobox functionality. Some parts, like the event handling will be a little tricky. We're happy to help support the engineering team when they're building this.

If you want to chat more about this, feel free to schedule some time in office hours.

**Comment by seanmateer** [comment]
Thanks both @TheSisb and @richbachman. Open to the same implementation with a Select field if it's easier, though I'm partial to the fancy styling of the combobox :) 

As a next step I am going to tag @alfredlucero just to get him up to speed on this discussion and we will follow up in implementation if we have any issues. You can probably resolve this issue in the meantime. Thank you!
